src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,70 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  isAlive() {
+    return this.health > 0;
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.die();
+    }
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
```

Comment:
[CODE PERFORMANCE] You are using `indexOf` and `splice` to remove the dead animal from the `alive` array. This can be inefficient because `indexOf` scans through the array to find the animal, and `splice` also scans through the array to remove it. It would be more efficient to use the `filter` method, which only scans through the array once.

Reply:
> [CODE PERFORMANCE] You are using `indexOf` and `splice` to remove the dead animal from the `alive` array. This can be inefficient because `indexOf` scans through the array to find the animal, and `splice` also scans through the array to remove it. It would be more efficient to use the `filter` method, which only scans through the array once.

Good point